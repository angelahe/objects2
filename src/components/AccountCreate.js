import React from 'react';
import closebtn from '../images/close_FFFFFF.png';
import "../styles/Styles140.css";

const AccountCreate = (props) => {

    console.log("props is ",props);

    function handleCloseClick() {
        console.log("in handleCloseClick");
    }

    function handleAddClick() {
        console.log("in handleAddClick function");
        const name = document.getElementById("acctName").value;
        const balance = Number(document.getElementById("acctBalance").value);
        props.createClicked(name, balance)
    }

    return(
        <div>
            <div className = "ItemBox AppHeader">
                <span className= "AddItem">Add Account</span>
                <button className="AppBtn" onClick={handleCloseClick}>
                    <img className="btnImg" src={closebtn} alt="Close"/>
                </button>
            </div>
            <div className = "ItemBox">
                <span className = "DetailText">Account Name:</span>
                <input className="InputText" id="acctName" autoFocus={true}></input><br/>
                <span className = "DetailText">Start Balance:</span>
                <input className = "InputText" id="acctBalance" type="number"></input><br/><br/>
                <button className="btnText" onClick={handleAddClick}>Create Account</button>

            </div>
        </div>
    )
}
export default AccountCreate