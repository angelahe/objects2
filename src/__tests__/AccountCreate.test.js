import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import AccountCreate from "../components/AccountCreate";
import renderer from "react-test-renderer";

xit('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AccountCreate />, div);
  ReactDOM.unmountComponentAtNode(div);
});

xit('renders without crashing', () => {
  shallow(<AccountCreate />);
});

xtest('AccountUI renders correctly', () => {
  const tree = renderer.create(<AccountCreate />).toJSON();
  expect(tree).toMatchSnapshot();
});