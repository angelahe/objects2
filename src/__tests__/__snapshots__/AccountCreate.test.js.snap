// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountUI renders correctly 1`] = `
<div>
  <div
    className="ItemBox AccountHeader"
  >
    <span
      className="AddAccount"
    >
      Add Account
    </span>
    <button
      className="AccountBtn"
      onClick={[Function]}
    >
      <img
        alt="Close"
        className="btnImg"
        src="close_FFFFFF.png"
      />
    </button>
  </div>
  <div
    className="ItemBox"
  >
    <span
      className="DetailText"
    >
      Account Name:
    </span>
    <input
      autoFocus={true}
      className="InputText"
      id="acctName"
    />
    <br />
    <span
      className="DetailText"
    >
      Start Balance:
    </span>
    <input
      className="InputText"
      id="acctBalance"
      type="number"
    />
    <br />
    <br />
    <button
      className="btnText"
      onClick={[Function]}
    >
      Create Account
    </button>
  </div>
</div>
`;
